sdf/final-project/technical.md

```diff
@@ -6,102 +6,138 @@ This is intended to be filled out by staff members of DPI Tech Apprenticeship pr
-- [ ] **Code Clarity**: Is the code easy to read and understand? Look for simple, straightforward implementations and avoid unnecessary complexity. Does the code include inline comments that explain "why" behind non-obvious logic? Over-commenting should be avoided; code should be self-explanatory wherever possible. Unused code should be deleted (not commented out).
-
-### Score (/5):
+## Code Hygiene (max: 7 points)
+- [ ] **Indentation**: Is the code consistently indented throughout the project?
+- [ ] **Naming Conventions**: Are naming conventions (e.g., variable names, method names, class names) clear, consistent, and descriptive?
+- [ ] **Casing Conventions**: Are casing conventions (e.g., camelCase, snake_case, PascalCase) consistent throughout the project?
+- [ ] **Layouts**: Is the code utilizing Rails' application.html.erb layout effectively, ensuring that it provides consistent and reusable templates for the application's views?
+- [ ] **Code Clarity**: Is the code easy to read and understand? Look for simple, straightforward implementations and avoid unnecessary complexity. 
+- [ ] **Comment Quality**: Does the code include inline comments that explain "why" behind non-obvious logic? Over-commenting should be avoided; code should be self-explanatory wherever possible. 
+- [ ] **Minimal Unused Code**: Unused code should be deleted (not commented out).
 
 ### Notes:
 
-## Frontend
-- [ ] **Semantic HTML**: Is the project making effective use of semantic HTML elements to structure the content, ensuring that it's both accessible and SEO-friendly?
-- [ ] **Styling**: Does the frontend employ CSS or CSS frameworks (like Bootstrap) for styling?
-- [ ] **Template Usage for Reusability**: Are partial templates used to promote code reusability and maintainability for common UI components (e.g., navigation bars, forms, lists)?
-- [ ] **Client-Side Interactivity**: Is JavaScript or JavaScript frameworks/libraries (e.g., jQuery) utilized to enhance the UI with dynamic content and interactive elements, reducing unnecessary page reloads?
-- [ ] **Rails View Helpers**: Does the project leverage Rails view helpers (link_to, form_with, etc.) to streamline the generation of view code and enforce best practices in form and link creation?
-
-### Bonus (max 2pts)
-- [ ] **Responsive Design**: It looks and works great on all different devices (seamless user experience across desktop, tablet, and mobile)
-- [ ] **Accessibility Considerations**: Are accessibility features (such as alt tags for images, ARIA roles) implemented to support users with disabilities and to comply with web accessibility standards?
-- [ ] **Enhanced Navigation and Feedback**: Are breadcrumbs used to enhance site navigation? Are styled flashes or toasts implemented to provide clear, user-friendly feedback? (in a partial)
+## Patterns of Enterprise Applications (max: 8 points)
```

Comment:
maybe add "event-driven architecture" (eg pub-sub) a common use-case would be something like action cable https://guides.rubyonrails.org/action_cable_overview.html